### PR TITLE
Refactor src and test MOI_wrappers for style and clarity

### PR DESCRIPTION
**Pull request by odow**

### CODE REVIEW

**Comment by odow** [comment]
I thought going through the code might help me spot any glaring errors. But the failures on mac seem sporadic. Local test failures seem random and are not reproducible.

File: src/MOI_wrapper.jl [hunk 1]
```diff
@@ -543,45 +548,66 @@ function vectorize_block(
...
+function MOI.get(
+    model::Optimizer,
+    attr::MOI.ConstraintPrimal,
+    ci::MOI.ConstraintIndex{
+        MOI.ScalarAffineFunction{Cdouble},
+        MOI.EqualTo{Cdouble},
+    },
+)
+    MOI.check_result_index_bounds(model, attr)
+    # TODO(odow): this isn't correct. In Ax = b, it should be Ax, not b.
```
Comment (blegat) [inline src/MOI_wrapper.jl:578]: Does removing this method work now that we have https://github.com/jump-dev/MathOptInterface.jl/pull/1674 ?

Comment (odow) [reply src/MOI_wrapper.jl:578]: Yes, we should be able to do something like that. I avoided making large changes in this PR though.

**Comment by odow** [comment]
Merging because I have a few follow-up PRs that change things other than style.

**Pull request merged**